scripts/ingests/BONES_archive/ingest_BONES_spectral_types.py

```diff
@@ -94,15 +92,17 @@ def extractADS(link):
         sp_regime = find_regime(source)
         ads = extractADS(source["ADS_LINK"])
         ref = find_publication(db=db, bibcode = ads)
-        #if publication not in db, ingest
-        try:
-            if ref[0] == False:
+        # if publication not in db, ingest
+        if ref[0] is False:
+            try:
                 ingest_publication(db = db, bibcode = ads)
-                pub_ingested +=1
-        except:
-            logger.warning("Find and ingest pattern didn't work" + ads)
-        ref = find_publication(db=db, bibcode=ads)
-        #ingest the spectral types
+                pub_ingested += 1
+                ref = find_publication(db=db, bibcode=ads)
+            except Exception as e:
+                logger.error("Find and ingest pattern didn't work" + ads)
+                raise e
```

Comment:
When you encounter an exception, you should raise the error so that the code stops and you can figure out what's wrong.